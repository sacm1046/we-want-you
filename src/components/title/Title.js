import "./style.scss";

export default function Title(props) {
  const { title, subTitle } = props;
  return (
    <>
      <h1 className="msg-header">{title}</h1>
      <h2 className="msg-sub-header">{subTitle}</h2>
    </>
  );
}
