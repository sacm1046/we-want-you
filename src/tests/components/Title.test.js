import React from "react";
import { shallow } from "enzyme";
import Title from "../../components/title/Title";

describe("Test para components/title/Title", () => {
  const title = "Título de prueba";
  const subTitle = "Sub título de prueba";
  const wrapper = shallow(<Title title={title} subTitle={subTitle} />);

  test("Se debe renderizar el componente Title", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Se debe mostrar el sub título enviado por props", () => {
    const foundSubTitle = wrapper.find("h2").text();
    expect(foundSubTitle).toBe(subTitle);
  });

  test("El título del componente debe recibir la props de title", () => {
    const foundTitle = wrapper.find("h1");
    expect(foundTitle.text().trim()).toBe(title);
  });

  test("El título del componente debe contener la clase msg-header", () => {
    const foundTitle = wrapper.find("h1");
    expect(foundTitle.prop("className")).toBe("msg-header");
  });
});
