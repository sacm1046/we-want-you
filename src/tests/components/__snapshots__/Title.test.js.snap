// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para components/title/Title Se debe renderizar el componente Title 1`] = `
<Fragment>
  <h1
    className="msg-header"
  >
    Título de prueba
  </h1>
  <h2
    className="msg-sub-header"
  >
    Sub título de prueba
  </h2>
</Fragment>
`;
